### DISCUSSION TITLE
Editor extension portability

### DISCUSSION THREAD

**Discussion by wesharper** [gitbutlerapp/gitbutler#5734] [Ideas]
I personally like the standalone app, but since there's a request for a VS Code extension, it got me thinking about portability to other editors like Neovim, Zed, Sublime, etc. and whether any APIs can be essentially reused as-is with a variety of clients in mind.

**Comment by Byron** [comment]
That's a very interesting thought - thanks for sharing!

When reading `Zed`, I think one would have to become a WASM plugin, which right now is hard to achieve natively as `gitoxide` for instance won't work in WASM mode. However, it WASM would only be used as integration layer for the backend binary which uses `tauri` to communicate to it, then everything that the GUI can do, the plugin could do.

The same *should* be true for a VSCode extension, which in theory should be able to host the frontend code as-is, with some glue that brings up the backend.
